robot_smach_states/src/robot_smach_states/util/designators/core.py

```diff
@@ -130,17 +158,11 @@ def __init__(self, initial_value=None, resolve_type=None, name=None):
                             "can use it")
 
         super(VariableDesignator, self).__init__(initial_value, resolve_type, name=name)
-
         self._current = initial_value
-
         self.writeable = VariableWriter(self)
 
     def _set_current_protected(self, value):
         resolve_type = self.resolve_type
-        try:
-            resolve_type = tuple(self.resolve_type)
```

Comment:
WTF was this?

Reply:
Likely caused by confusion about the semantics of "[<type>]" vs "<type>" in `resolve_type`, although I didn't check history.

Storing a non-type value in a variable that should contain a type is quite confusing, the proper solution would be to make 2 classes, one for expressing a value for a singular value, and one for expressing a value for a list of items.

Reply:
Yes, I've been pondering to split the VariableDesignator into one for collections and one for singular values. But That's for later I'd say